### PR TITLE
Feature/basic go to pos

### PR DESCRIPTION
**Pull request by RobZelluf**
- [ ] The tests are passing

### Summary
Make GoToPos a skill with just one functions: go to a position that is set in onInitialize.
BasicGoToBos is renamed to GTPSpecial, and takes a "type" as parameter. GTPSpecial is a child of GoToPos that overrides the onInitialize function to set a custom targetPos.

### Comments for the reviewer
I also changed all the GoToPosses in all roles and tactics, but this might need some double checking

**functionality**  
- Not all tactics and roles where individually tested to see if they still work

### CODE REVIEW

File: roboteam_ai/src/jsons/roles/DemoKeeper.json [hunk 1]
```diff
@@ -1 +1,232 @@
...
+            "title": "DribbleRotate"
+          }
+        },
+        "root": "o3jcbqg12iz4ba9",
+        "title": "DemoKeeper"
+      }
+    ]
+  },
+  "name": "DemoKeeper"
+}
```
Comment (mrlukasbos) [inline roboteam_ai/src/jsons/roles/DemoKeeper.json]: remove the backup_jsons. we have git.

File: roboteam_ai/src/skills/GoToPos.cpp [hunk 2]
```diff
@@ -14,134 +14,51 @@ GoToPos::GoToPos(string name, bt::Blackboard::Ptr blackboard)
...
-        distanceBehindBall = properties->getDouble("distanceBehindBall");
-    } else {
-        distanceBehindBall = 0.2;
+    if (properties->hasVector2("targetPos")) {
+        targetPos = properties->getVector2("targetPos");
     }
 
-    if (properties->hasVector2("Position")) {
-        targetPos = properties->getVector2("Position");
+    if (properties->hasDouble("maxVel")) {
```
Comment (mrlukasbos) [inline roboteam_ai/src/skills/GoToPos.cpp]: is position being replaced by maxvel here?

File: roboteam_ai/src/skills/GoToPos.cpp [hunk 3]
```diff
@@ -14,134 +14,51 @@ GoToPos::GoToPos(string name, bt::Blackboard::Ptr blackboard)
...
     roboteam_msgs::RobotCommand command;
     command.id = robot->id;
-    command.use_angle = 1;
-    command.w = static_cast<float>(deltaPos.angle());
 
-    command.x_vel = 0;
-    command.y_vel = 0;
+    std::cout << targetPos << std::endl;
+    control::PosVelAngle pva = goToPos.goToPos(robot, targetPos, control::PosControlType::NUMERIC_TREES);
+    std::cout << pva.vel << std::endl;
```
Comment (mrlukasbos) [inline roboteam_ai/src/skills/GoToPos.cpp]: remove the std::cout statements

File: roboteam_ai/test/SkillTests/GoToPosTest.cpp [hunk 4]
```diff
@@ -45,6 +43,13 @@ TEST(GoToPos, GoToPosTest) {
 
     EXPECT_EQ(goToPos.update(), bt::Leaf::Status::Running);
 
+    robot.pos.x = 5.0;
+    robot.pos.y = 6.0;
+
+    worldMsg.us[0] = robot;
+    rtt::ai::World::set_world(worldMsg);
+
+    EXPECT_EQ(goToPos.update(), bt::Leaf::Status::Success);
```
Comment (mrlukasbos) [inline roboteam_ai/test/SkillTests/GoToPosTest.cpp]: Maybe you can at least test that the target positions are coming through properly

File: roboteam_ai/src/jsons/roles/DemoKeeper.json (hunk 1 above)
Comment (RobZelluf) [reply roboteam_ai/src/jsons/roles/DemoKeeper.json]: already done

File: roboteam_ai/src/skills/GoToPos.cpp (hunk 2 above)
Comment (RobZelluf) [reply roboteam_ai/src/skills/GoToPos.cpp]: No, position is being replaced by targetPos and maxVel is added

File: roboteam_ai/src/skills/GoToPos.cpp (hunk 3 above)
Comment (RobZelluf) [reply roboteam_ai/src/skills/GoToPos.cpp]: Done

File: roboteam_ai/test/SkillTests/GoToPosTest.cpp (hunk 4 above)
Comment (RobZelluf) [reply roboteam_ai/test/SkillTests/GoToPosTest.cpp]: Yes

File: roboteam_ai/src/utilities/Constants.h [hunk 5]
```diff
@@ -52,6 +52,8 @@ class Constants {
     static double CLOSE_TO_BORDER_DISTANCE()    { return 1.2 * ROBOT_RADIUS(); };
 
     //skills
+    static double GOTOPOS_ERROR_MARGIN()        { return 0.01;} ;
```
Comment (Barisimre) [inline roboteam_ai/src/utilities/Constants.h]: This is ambitious

File: roboteam_ai/src/skills/GTPSpecial.h [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+#include "GoToPos.h"
+#include "../interface/drawer.h"
+
+
+namespace rtt {
+namespace ai {
+
+class GTPSpecial : public GoToPos {
+    private:
+        enum Type {
```
Comment (Barisimre) [inline roboteam_ai/src/skills/GTPSpecial.h]: Add the default this we talked about

**Review by Barisimre — CHANGES_REQUESTED** [review]
Just fix em then we merge

File: roboteam_ai/src/utilities/Constants.h (hunk 5 above)
Comment (RobZelluf) [reply roboteam_ai/src/utilities/Constants.h]: Done

File: roboteam_ai/src/skills/GTPSpecial.h (hunk 6 above)
Comment (RobZelluf) [reply roboteam_ai/src/skills/GTPSpecial.h]: Yepyepyep

**Review by Barisimre — APPROVED** [review]

**Pull request merged**